src/interactive/azext_interactive/azclishell/recommendation.py

```diff
@@ -0,0 +1,392 @@
+        self.on_recommendation_prepared = lambda: None
+        self.default_recommendations = {
+            'help': 'Get help message of Azure CLI',
+            'init': 'Set Azure CLI global configurations interactively',
+            'next': 'Recommend the possible next set of commands to take'
+        }
+        self.executing_command = None
+
+    @property
+    def enabled(self):
+        """Whether recommender is enabled in global config"""
+        return self.cli_ctx.config.getboolean("interactive", "enable_recommender", fallback=True)
+
+    def _feedback_command(self, command):
+        """Send user's command choice in recommendations to telemetry."""
+        if self.cur_thread and not self.cur_thread.is_alive():
+            recommendations = self.cur_thread.result
+            trigger_commands = [cmd['command'] for cmd in self.cur_thread.command_history[-2:]]
+            processed_exception = self.cur_thread.processed_exception
+            if not recommendations or not command:
+                send_feedback(-1, trigger_commands, processed_exception, recommendations, rec=None)
+                return
+            for idx, rec in enumerate(recommendations):
+                if rec['type'] != RecommendType.Command:
+                    continue
+                if re.sub(r'^az ', '', re.sub(r'\s+', ' ', command)).strip().startswith(rec['command']):
+                    send_feedback(f'a{idx+1}', trigger_commands, processed_exception, recommendations, rec)
+                    return
+            send_feedback(0, trigger_commands, processed_exception, recommendations, rec=None)
+
+    def feedback_scenario(self, scenario_idx, scenario=None):
+        """Send user's command choice in recommendations to telemetry.
+
+        :param scenario_idx: idx of the scenario in scenario recommendations list
+        :param scenario: selected scenario item
+        """
+        if self.cur_thread and not self.cur_thread.is_alive():
+            recommendations = self.cur_thread.result
+            trigger_commands = [cmd['command'] for cmd in self.cur_thread.command_history[-2:]]
+            processed_exception = self.cur_thread.processed_exception
+            if not recommendations or not scenario:
+                send_feedback(-1, trigger_commands, processed_exception, recommendations, rec=None)
+            else:
+                send_feedback(f'b{scenario_idx+1}', trigger_commands, processed_exception, recommendations, rec=scenario)
+
+    def update_executing(self, cmd, feedback=True):
+        """Update executing command info, and prefetch the recommendation result as if the execution is successful
+
+        :param cmd: The command that would be executed soon
+        :param feedback: whether send feedback to telemetry using this command as actual execution vs
+        previous recommendation
+        """
+        if not self.enabled:
+            return
+        command = re.sub('^az *', '', cmd).split('-', 1)[0].strip()
```

Comment:
Please add the comments and example for the regular expression

Reply:
Done